Yank/tests/test_experiment.py

```diff
@@ -200,6 +205,88 @@ def get_template_script(output_dir='.'):
     return yank_load(template_script)
 
 
+def get_functionality_script(output_directory=',', number_of_iter=0, experiment_repeats=1, number_nan_repeats=0):
+    """
+    A computationally simple pre-setup system which can be loaded to manipulate a formal experiment
+    Should not be used for scientific testing per-se, but can be used to test functional components of experiment
+
+    Parameters
+    ==========
+    output_directory : str, Optional
+        Output directory to set in script
+    number_of_iter : int Optional, Default: 1
+        Number of iterations to run
+    experiment_repeats : int, Optional, Default: 1
+        Number of times the experiment is repeated in a "experiments" header
+    number_nan_repeats : int, Optional, Default: 0
+        Number of times the experiment with a NaN is repeated, this will be added to the end of the stack
+    """
+    paths = examples_paths()
+    template_script = """
+    ---
+    options:
+      minimize: no
+      verbose: yes
+      output_dir: {output_directory}
+      number_of_iterations: {number_of_iter}
+      nsteps_per_iteration: 10
+      temperature: 300*kelvin
+      pressure: null
+      anisotropic_dispersion_cutoff: null
+
+    solvents:
+      vacuum:
+        nonbonded_method: NoCutoff
+
+    systems:
+      premade:
+        phase1_path: {boxless_path}
+        phase2_path: {boxless_path}
+        ligand_dsl: resname ene
+        solvent: vacuum
+      premade_nan:
+        phase1_path: {nan_path}
+        phase2_path: {nan_path}
+        ligand_dsl: resname ene
+        solvent: vacuum
+
+    protocols:
+      absolute-binding:
+        complex:
+          alchemical_path:
+            lambda_electrostatics: [0.0, 0.0]
+            lambda_sterics:        [0.0, 0.0]
+        solvent:
+          alchemical_path:
+            lambda_electrostatics: [1.0, 1.0]
+            lambda_sterics:        [1.0, 1.0]
+
+    the_exp:
+      system: premade
+      protocol: absolute-binding
+      restraint:
+        type: FlatBottom
+    the_nan_exp:
+      system: premade_nan
+      protocol: absolute-binding
+      restraint:
+        type: FlatBottom
+
+    experiments: [{repeating}]
+
+    """
+    repeating_string = ', '.join(['the_exp'] * experiment_repeats)
+    repeating_nan_string = ', '.join(['the_nan_exp'] * number_nan_repeats)
```

Comment:
I believe this has no effect. The experiments are collected into a dictionary inside `ExperimentBuilder`, which means that there will be only 1 `the_exp` and 1 `the_nan_exp` regardless of how many times you specify them in the `experiments` list.